File: src/test/unit/LectureRegistrationApplicationTest.java
package test.unit;

import static org.junit.Assert.assertEquals;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import org.junit.Test;

import Debt_LRA_Transcript.Debt;
import Debt_LRA_Transcript.LectureRegistrationApplication;
import Debt_LRA_Transcript.Transcript;
import Enums.ApprovalState;
import IDs.InstructorID;
import IDs.SessionID;
import IDs.StudentID;
import lecture.LectureSession;
import lecture.Semester;
import person.Advisor;
import person.Student;

public class LectureRegistrationApplicationTest {

	@Test
	public void test_LRA_default_constructor() {
		
		Map<LectureSession, ApprovalState> testSessionsSentForApproval = new HashMap<LectureSession, ApprovalState>();
		SessionID testID = new SessionID(123);
		
		LectureSession testLectureSession = new LectureSession(testID , null , null, null, null, null);
		
		Semester testSemester = new Semester(null, 1, 2, 2.1 , 1.2);
		Semester testSemester2 = new Semester(null, 2, 3, 3.1, 1.3);
		List<Semester> testListOfSemester = new ArrayList<>();
		testListOfSemester.add(testSemester);
		testListOfSemester.add(testSemester2);
		
		Transcript testTranscript = new Transcript(null, testListOfSemester, 
				3.4, 23, 13, 2.3);
		
		StudentID testId = new StudentID(000,000,000);
		InstructorID testInstructorID = new InstructorID(123, 456);
		
		Advisor testAdvisor = new Advisor("firstName", "lastName", testInstructorID, null, null, null, null, null);
		
		Student testStudent = new Student("testFName", "testLName",
				testId, null, testTranscript, null);
		
		LectureRegistrationApplication testLRA = new LectureRegistrationApplication(testSessionsSentForApproval, testAdvisor, testStudent);
		
		
		testLRA.setAdvisor(testAdvisor);
		assertEquals(testAdvisor, testLRA.getAdvisor());
		
		testLRA.setStudent(testStudent);
		assertEquals(testStudent, testLRA.getStudent());
		
		testLRA.setSessionsSentForApproval(testSessionsSentForApproval);
		assertEquals(testSessionsSentForApproval, testLRA.getSessionsSentForApproval());
		
		
	}

}

File: pythonCode/LectureRegistrationApplicationTest.py
#!/usr/bin/env python
from Advisor import Advisor;
from Session import Session
from Student import Student
from LectureRegistrationApplication import LectureRegistrationApplication
import unittest
class LectureRegistrationApplicationTest(unittest.TestCase):
    # generated source for class LectureRegistrationApplicationTest 
    def test_LectureRegistrationApplicationTest_Advisor(self):
        # generated source for method test_LectureRegistrationApplicationTest_Advisor 
        testAdvisor = Advisor()
        testAdvisor.setFirstName("testFName")
        testLectureRegistrationApplicationTest = LectureRegistrationApplication()
        testLectureRegistrationApplicationTest.setAdvisor(testAdvisor)
        self.assertEqual(testAdvisor, testLectureRegistrationApplicationTest.getAdvisor())
    def test_LectureRegistrationApplicationTest_Student(self):
        # generated source for method test_LectureRegistrationApplicationTest_Student 
        testStudent = Student()
        testStudent.setFirstName("testFName")
        testLectureRegistrationApplicationTest = LectureRegistrationApplication()
        testLectureRegistrationApplicationTest.setStudent(testStudent)
        self.assertEqual(testStudent, testLectureRegistrationApplicationTest.getStudent())
    def test_LectureRegistrationApplicationTest_SessionsSentForApproval(self):
        # generated source for method test_LectureRegistrationApplicationTest_SessionsSentForApproval
        testSession = Session()
        testSessionsSentForApproval = list()
        testSessionsSentForApproval.append(testSession)
        testLectureRegistrationApplicationTest = LectureRegistrationApplication()
        testLectureRegistrationApplicationTest.setSsessionsSentForApproval(testSessionsSentForApproval)
        self.assertEqual(testSessionsSentForApproval, testLectureRegistrationApplicationTest.getSessionsSentForApproval())
if __name__ == '__main__':
    unittest.main()